src/test/java/io/camunda/zeebe/process/test/multithread/MultiThreadTest.java

```diff
@@ -0,0 +1,69 @@
+package io.camunda.zeebe.process.test.multithread;
+
+import static io.camunda.zeebe.process.test.assertions.BpmnAssert.assertThat;
+import static io.camunda.zeebe.process.test.util.Utilities.deployProcess;
+import static io.camunda.zeebe.process.test.util.Utilities.startProcessInstance;
+
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.client.api.response.ProcessInstanceEvent;
+import io.camunda.zeebe.process.test.RecordStreamSourceStore;
+import io.camunda.zeebe.process.test.extensions.ZeebeProcessTest;
+import io.camunda.zeebe.process.test.testengine.InMemoryEngine;
+import io.camunda.zeebe.process.test.testengine.RecordStreamSource;
+import io.camunda.zeebe.process.test.util.Utilities.ProcessPackStartEndEvent;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+@ZeebeProcessTest
+public class MultiThreadTest {
```

Comment:
:thought_balloon: I am of two minds about this test. On the one hand I am glad there is a test at all. But this test looks like it might be a flaky test. It feels too short, and I could imagine even with the wrong implementation it will sometimes pass and sometimes fail. I might be wrong though. 

Also, I don't really have a better idea.

There is the option of using _jqwik_ and having a number of runs with a random number of runners and each looping a random number of times with a random operation. But at the same time this feels like overkill. 

Still, if you feel like making this more solid, you could have a look at: https://github.com/camunda-cloud/zeebe/blob/develop/atomix/cluster/src/test/java/io/atomix/raft/RandomizedRaftTest.java or https://github.com/camunda-cloud/zeebe/blob/develop/broker/src/test/java/io/camunda/zeebe/broker/system/partitions/impl/RandomizedPartitionTransitionTest.java

None of them are clean examples of property based tests. I am not saying you need to change this test. But if you either feel uncomfortable about this test and/or are curious about property based testing, you might give this a go

Reply:
I understand your concern. However, the test didn't seem flaky when I was playing around with it. Without the fix it failed 100% of the runs, and with the fix I haven't seen it fail yet.

I would propose to keep it as is and if it turns out to be flaky and becomes a problem we can fix it at that time.